plasmapy/analysis/fit_functions.py

```diff
@@ -0,0 +1,1141 @@
+
+    def __str__(self):
+        return f"f(x) = A exp(alpha x)"
+
+    def func(self, x, a, alpha):
+        """
+        The fit function, a exponential function.
+
+        .. math::
+
+            f(x) = A \\, e^{\\alpha \\, x}
+
+        where :math:`A` and :math:`\\alpha` are real constants and :math:`x`
+        is the independent variable.
+
+        Parameters
+        ----------
+        x: array_like
+            Independent variable.
+
+        a: float
+            value for the exponential "normalization" constant, :math:`A`
+
+        alpha: float
+            value for the growth constant, :math:`\\alpha`
+
+        Returns
+        -------
+        y: array_like
+            dependent variables corresponding to :math:`x`
+
+        """
+        x = self._check_x(x)
+        self._check_params(a, alpha)
+
+        return a * np.exp(alpha * x)
+
+    def func_err(self, x, x_err=None, rety=False):
+        """
+        Calculate dependent variable uncertainties :math:`\\delta y` for
+        dependent variables :math:`y=f(x)`.
+
+        .. math::
+
+            \\left( \\frac{\\delta y}{|y|} \\right)^2 =
+                \\left( \\frac{\\delta A}{A} \\right)^2
+                + (x \\, \\delta \\alpha)^2
+                + (\\alpha \\, \\delta x)^2
+
+        Parameters
+        ----------
+        x: array_like
+            Independent variables to be passed to the fit function.
+
+        x_err: array_like, optional
+            Errors associated with the independent variables `x`.  Must be of
+            size one or equal to the size of `x`.
+
+        rety: bool
+            Set `True` to also return the associated dependent variables
+            :math:`y = f(x)`.
+
+        Returns
+        -------
+        err: `numpy.ndarray`
+            The calculated uncertainties :math:`\\delta y` of the dependent
+            variables (:math:`y = f(x)`) of the independent variables `x`.
+
+        y: `numpy.ndarray`, optional
+            (if `rety = True`) The associated dependent variables
+            :math:`y = f(x)`.
+        """
+        x = self._check_x(x)
+        if x_err is not None:
+            x_err = self._check_x(x_err)
+
+            if x_err.shape == ():
+                pass
+            elif x_err.shape != x.shape:
+                raise ValueError(
+                    f"x_err shape {x_err.shape} must be equal the shape of "
+                    f"x {x.shape}."
+                )
+
+        a, alpha = self.params
+        a_err, alpha_err = self.param_errors
+        y = self.func(x, a, alpha)
+
+        a_term = (a_err / a) ** 2
+        alpha_term = (x * alpha_err) ** 2
+
+        err = a_term + alpha_term
+
+        if x_err is not None:
+            x_term = (alpha * x_err) ** 2
+            err += x_term
+
+        err = np.abs(y) * np.sqrt(err)
+
+        if rety:
+            return err, y
+
+        return err
+
+    @property
+    def latex_str(self) -> str:
+        return fr"A \, \exp(\alpha \, x)"
```

Comment:
I'm not so much concerned with someone copying them to a paper.  If someone does, I'm assuming they're going to reformat no matter what.  No one puts something into LaTeX without spending at least half an hour trying to format it.

These are predominately here to work with `matplotlib` plotting.  I would value it rendering nicely there over someone copying it to a LaTex file.  In `matplotlib` it would  not be possible to adjust spacing, unlike pasting it in a LaTeX file.  

Here's an example with and without the `\,`

![image](https://user-images.githubusercontent.com/29869348/96326550-73862d00-0fe6-11eb-9696-e934bdc3456d.png)

Honestly, I think it looks too smushed together without the `\,` and more difficult to read.  If it's a real sticking point, then I can concede on all of them excepted for `A \, \exp()`.

Reply:
> I'm not so much concerned with someone copying them to a paper. If someone does, I'm assuming they're going to reformat no matter what. No one puts something into LaTeX without spending at least half an hour trying to format it.

> ![image](https://user-images.githubusercontent.com/29869348/96326550-73862d00-0fe6-11eb-9696-e934bdc3456d.png)

> Honestly, I think it looks too smushed together without the `\,` and more difficult to read. If it's a real sticking point, then I can concede on all of them excepted for `A \, \exp()`.

I think this is my most preferred option.  Apart from separating `A` and `\exp`, I think the formatting for the second one looks better, but that's my own opinion.

Brainstorming some additional possibilities...
 - Change `latex_str` to `_latex_str` so that it's not part of the public API, and keep the formatting as is.  I would support this iff we only anticipate using it internally.  
 - Have a private variable `_latex_str_with_extra_spaces`. 
 - Change the name to reflect that it's for use in plotting, and keep the formatting.

In any case, I'm happy to go with your judgement.

Reply:
A grumpy aside is that using LaTeX to represent math (as is standard practice) loses the semantic information.  Like...`mx` will place the variables together so we know it's multiplication when we read it, but the code loses the semantic meaning that there's an implicit multiplication, which makes it harder to do text-to-speech conversion of equations for people who depend on screen reader technology.  That's why I want to eventually look into implementing MathML and mathjax across PlasmaPy...but that's just ever so slightly outside of scope of this PR.